projects/optic/src/commands/capture/capture.ts

```diff
@@ -197,43 +202,71 @@ const getCaptureAction =
     }
 
     // make requests
-    const concurrency = captureConfig.config?.request_concurrency || 5;
-    const requests = makeRequests(
-      captureConfig.requests,
-      proxyUrl,
-      concurrency
-    );
-
     const captures = new GroupedCaptures(
       trafficDirectory,
       getEndpointsFromSpec(spec.jsonLike)
     );
-    const harEntries = HarEntries.fromProxyInteractions(proxyInteractions);
 
-    await Promise.all(requests)
-      .catch((error) => {
-        console.error(error);
-      })
-      .finally(() => {
-        // stop the proxy
-        sourcesController.abort();
-        // stop the app server
-        if (!options.serverOverride && captureConfig.server.command) {
-          process.kill(-app.pid!);
-        }
-      });
+    let requestsPromise: Promise<any> = Promise.resolve();
+    let reqCmdPromise: Promise<void> = Promise.resolve();
+    try {
+      if (captureConfig.requests) {
+        const requests = makeRequests(
+          captureConfig.requests,
+          proxyUrl,
+          captureConfig.config?.request_concurrency || 5
+        );
+        requestsPromise = Promise.all(requests);
+      }
+
+      if (captureConfig.requests_command) {
+        const cmd = commandSplitter(captureConfig.requests_command.command);
+        const proxyVar =
+          captureConfig.requests_command.proxy_variable || 'OPTIC_PROXY';
+        process.env[proxyVar] = proxyUrl;
+        const reqCmd = spawn(cmd.cmd, cmd.args, {
+          detached: true,
+          shell: true,
+        });
+
+        reqCmdPromise = new Promise((resolve, reject) => {
+          reqCmd.on('exit', (code) => {
+            if (code === 0) {
+              resolve();
+            } else {
+              reject();
+            }
+          });
+        });
+        reqCmd.stderr.on('data', (data) => {
+          logger.error(data.toString());
+        });
+      }
+    } catch (error) {
+      logger.error(error);
+    } finally {
+      await Promise.all([requestsPromise, reqCmdPromise]);
+      sourcesController.abort();
+      if (app) {
+        process.kill(-app.pid!);
+      }
+    }
 
+    const harEntries = HarEntries.fromProxyInteractions(proxyInteractions);
+    let count = 0;
     for await (const har of harEntries) {
+      count++;
       captures.addHar(har);
+      logger.debug(`Captured ${har.request.url}`);
     }
     await captures.writeHarFiles();
 
+    logger.info(
+      `${count} requests captured. Run \`optic update --all\` to document updates.`
```

Comment:
```suggestion
      `${count} requests captured`
```

We're removing the need to run `optic update` after this. I'm also not sure that rendering the request count captured is super important but we can figure that out later. We might want to instead show the endpoints we've gotten requests on (and # of unmatched requests)

Reply:
> I'm also not sure that rendering the request count captured is super important but we can figure that out later

yeah, im cool to remove it later. it's mostly useful for me now to make sure something happens. otherwise i `LOG_LEVEL=debug` and its a bit verbose